### PR TITLE
Backend with user and quiz management

### PR DESCRIPTION
**Pull request by Rupamdas832**
-Login/signup user
-Add a new Quiz and questions for it.
-Update user's score and accuracy after every quiz played

### CODE REVIEW

File: QuizData/quiz.model.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                type: String,
+              },
+                points: {
+                type: Number,
+              },
+                negativePoints: {
+                type: Number,
+              },
+              options: [{value: String, isCorrect: Boolean}]
+  }]
```
Comment (0xRishav) [inline QuizData/quiz.model.js:36]: adding timestamps in models is a good practice

**Review by 0xRishav — COMMENTED** [review]
Hey @Rupamdas832, the code is very well written couldn't find much. Just suggested adding timestamps in models

File: Database/DBconnect.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const mongoose = require("mongoose")
+
+const mySecret = process.env['MongoDBCredentials']
+
+const uri = mySecret;
```
Comment (neharajesh) [inline Database/DBconnect.js]: `mySecret` could be removed, you can just set mongodb credentials to `uri`

File: Database/DBconnect.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+const initializeDBconnection = async () => {
+  try{
+    const response = await mongoose.connect(uri, {
+      useNewUrlParser: true, 
+      useUnifiedTopology: true
+    })
+    if(response){
+      console.log("MONGOOSE Connected successfuly")
+    }
```
Comment (neharajesh) [inline Database/DBconnect.js]: mongoose.connect will always return a response. If you want to check if mongoose has connected, you can try checking the `mongoose.connection.readyState` property. https://masteringjs.io/tutorials/mongoose/connection-status

File: QuizData/uploadQuiz.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    }catch(error){
+      res.status(500).json({success: false, message: "Sorry! couldn't add quiz. Retry...", error})
+    }
+  })
+
+  router.route("/:quizId")
+    .post(async (req,res) => {
+      const {question,points,negativePoints,value1,isCorrect1, value2,isCorrect2, value3, isCorrect3, value4, isCorrect4} = req.body;
+
+      let {quiz} = req;
```
Comment (neharajesh) [inline QuizData/uploadQuiz.js:72]: Since you've already found the value in middleware, this could be const

File: Users/updateUser.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      req.user = findUser
+      next();
+    }catch(error){
+      res.status(400).json({success: false, message: "Something went wrong"})
+    }
+  })
+
+router.route("/:userId")
+  .post(async(req,res) => {
+    let {user} = req;
```
Comment (neharajesh) [inline Users/updateUser.js:22]: this could be const too

**Review by neharajesh — COMMENTED** [review]
Hey Rupam,
Really well written code! Was interesting to review! 
I've left a few suggestions, hope they help!

File: QuizData/uploadQuiz.js (hunk 4 above)
Comment (Rupamdas832) [reply QuizData/uploadQuiz.js:72]: But I have to change its value below it.

File: Database/DBconnect.js (hunk 3 above)
Comment (Rupamdas832) [reply Database/DBconnect.js]: Thanks. I will look into it.

File: QuizData/quiz.model.js (hunk 1 above)
Comment (Rupamdas832) [reply QuizData/quiz.model.js:36]: Ya good point. Thanks I will add

File: Database/DBconnect.js (hunk 2 above)
Comment (Rupamdas832) [reply Database/DBconnect.js]: Ya wrote an extra code. Thanks.

**Pull request merged**